shame_command.py

```diff
@@ -0,0 +1,80 @@
+import configparser
+import logging
+import os
+import aiohttp
+import discord
+from discord.ext import commands
+
+logger = logging.getLogger(__name__)
+
+# Load the config file
+config_path = os.path.join(os.path.dirname(__file__), "settings.cfg")
+config = configparser.ConfigParser()
+config.read(config_path)
+
+TODOIST_API = "https://api.todoist.com/rest/v2/tasks"
+
+
+# Retrieve Todoist API Key for the email from the config file
+def get_todoist_token(email: str) -> str:
+    config.read(config_path)
+    if email in config["TODOIST_KEY_BY_EMAIL"]:
+        return config["TODOIST_KEY_BY_EMAIL"][email]
+    else:
+        return None
+
+
+# Get tasks from Todoist API with 'shame' label using aiohttp
+async def get_shame_tasks(todoist_token: str, session: aiohttp.ClientSession) -> list:
+    headers = {"Authorization": f"Bearer {todoist_token}"}
+    params = {"filter": "label:shame"}
+
+    async with session.get(TODOIST_API, headers=headers, params=params) as response:
+        if response.status != 200:
+            logger.error(f"Failed to retrieve tasks: {response.status}")
+            return []
+        return await response.json()
+
+
+# Discord bot command to shame the user
+@commands.command(name="shame")
+async def shame(interaction: discord.Interaction, user_to_shame: discord.Member):
+    config.read(config_path)
+    email = None
+
+    # Find the email associated with the user
+    for stored_email, discord_id in config["DISCORD_ID_BY_EMAIL"].items():
+        if discord_id == str(user_to_shame.id):
+            email = stored_email
+            break
```

Comment:
There is a nice pattern to find the first predicate match:
https://stackoverflow.com/a/8534381

Reply:
I'm not entirely convinced that the pattern is more clear than writing the change out, but I don't feel too strongly one way or the other